src/v/cluster/tx_gateway_frontend.h

```diff
@@ -39,11 +41,13 @@ class tx_gateway_frontend final
       ss::sharded<cluster::rm_partition_frontend>&,
       ss::sharded<features::feature_table>&,
       ss::sharded<cluster::tm_stm_cache_manager>&,
-      ss::sharded<kafka::coordinator_ntp_mapper>& tx_coordinator_ntp_mapper);
+      ss::sharded<cluster::tx_coordinator_mapper>& tx_coordinator_ntp_mapper);
 
     ss::future<std::optional<model::node_id>>
       find_coordinator(kafka::transactional_id);
-    std::optional<model::ntp> get_ntp(kafka::transactional_id);
+    ss::future<std::optional<model::ntp>> get_ntp(kafka::transactional_id);
+    ss::future<checked<bool, tx_errc>>
```

Comment:
it's fine to have just bool there, on false we return not coordinator and we may return false on any error